### PR TITLE
Update line descriptor samples

### PR DESCRIPTION
**Pull request by sturkmen72**
see http://answers.opencv.org/question/169646

### CODE REVIEW

File: modules/xfeatures2d/samples/shape_transformation.cpp [hunk 1]
```diff
@@ -2,10 +2,10 @@
  * shape_context.cpp -- Shape context demo for shape matching
  */
 #include <iostream>
+#include "opencv2/shape.hpp"
```
Comment (sovrasov) [inline modules/xfeatures2d/samples/shape_transformation.cpp]: Why has it been moved outside of `#ifdef HAVE_OPENCV_SHAPE`?

Comment (sturkmen72) [inline modules/xfeatures2d/samples/shape_transformation.cpp]: @sovrasov we need to determine `HAVE_OPENCV_SHAPE` is defined or not. in actual code `HAVE_OPENCV_SHAPE` is always undefined. by this change it works as it should be

Comment (alalek) [inline modules/xfeatures2d/samples/shape_transformation.cpp]: Use `#include "opencv2/opencv_modules.hpp"` for proper `HAVE_OPENCV_SHAPE` definition.

**Comment by sovrasov** [comment]
:+1:

**Comment by sturkmen72** [comment]
thanks, i will rebase commits later asap

**Comment by sturkmen72** [comment]
Commits are rebased

**Pull request merged**